src/applications/disability-benefits/526EZ/components/AsyncDisplayWidget.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/formation/AlertBox';
+
+const PENDING = 'PENDING';
+const RESOLVED = 'RESOLVED';
+const REJECTED = 'REJECTED';
+
+
+/**
+ * Handles the various display statuses when calling an asynchronous function.
+ */
+export default class AsyncDisplayWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      promiseState: PENDING
+    };
+
+    // Make sure we configured it correctly
+    // StringField passes 'ui:options' as the options prop
+    const uiOptions = props.options;
+    if (!uiOptions) {
+      throw new Error('No ui:options supplied to AsyncDisplayWidget.');
+    }
+
+    if (typeof uiOptions.viewComponent !== 'function') {
+      throw new Error('AsyncDisplayWidget requires viewComponent in ui:options to be a React component.');
```

Comment:
This seems very helpful, how do we decide to leverage these errors vs. rely on proptypes? Also, what is the best approach to graceful failure in these cases?

Reply:
I decided to throw errors for these cases because the widget is _going_ to throw another error if these things are wrong, but the error it'll throw will probably not be super helpful. I think we really shouldn't fail gracefully in these cases because it's a programmer error and needs to be fixed in the form config.